Update theme for menus

3Simplex opened the pull request:
## Describe your changes
Menu theme colors adjusted to produce softer look while maintaining current scheme.

To achieve this:
Assigned property names for each color by the object to which they apply. (Ideally easy to understand without needing to trace)
- "menuBorderColor" (the color for the border of the comboBox & context menus)
- "menuBackgroundColor" (the color for the background of the comboBox & context menus)
- "menuHighlightColor" (the color for the highlight of the comboBox & context menus)

Other:
- The previous color properties no longer apply to other separate objects and were commented out with notes. (To be removed?)
- Fix for server text color in chatview. (tested)

## Issue ticket number and link

## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] I have added thorough documentation for my code.
- [ ] I have tagged PR with relevant project labels. I acknowledge that a PR without labels may be dismissed.
- [ ] If this PR addresses a bug, I have provided both a screenshot/video of the original bug and the working solution.

## Demo
<!-- Screenshots or video of new or updated code changes !-->

### Steps to Reproduce
<!-- Steps to reproduce demo !-->

## Notes
<!-- Any other relevant information to include about PR !-->

<!-- ELLIPSIS_HIDDEN -->

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 2a9ef8e696791ccb8729134ad2609903f33712cd  | 
|--------|--------|

### Summary:
Updated theme colors for menus and context menus with new color properties for a softer look in `gpt4all-chat/qml/ChatView.qml`, `gpt4all-chat/qml/MyMenu.qml`, and `gpt4all-chat/qml/MyMenuItem.qml`.

**Key points**:
- Updated theme colors for menus in `gpt4all-chat/qml/ChatView.qml`, `gpt4all-chat/qml/MyMenu.qml`, and `gpt4all-chat/qml/MyMenuItem.qml`.
- Introduced new color properties in `gpt4all-chat/qml/Theme.qml`:
  - `chatViewModelListFrameColor`
  - `chatViewModelListBgColor`
  - `chatViewModelListHighlightColor`
  - `contextMenuFrameColor`
  - `contextMenuBgColor`
  - `contextMenuHighlightColor`
- Applied new color properties to relevant UI components for a softer look while maintaining the current scheme.


----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)


<!-- ELLIPSIS_HIDDEN -->

manyoso commented on gpt4all-chat/qml/Theme.qml:
```diff
@@ -1164,6 +1164,50 @@ QtObject {
         }
     }
 
+    property color chatViewModelListFrameColor: {
```
Maybe this could just be MenuFrameColor ?

manyoso commented on gpt4all-chat/qml/Theme.qml:
```diff
@@ -1164,6 +1164,50 @@ QtObject {
+        }
+    }
+
+    property color chatViewModelListBgColor: {
```
MenuBackgroundColor?

manyoso commented on gpt4all-chat/qml/Theme.qml:
```diff
@@ -1164,6 +1164,50 @@ QtObject {
+        }
+    }
+
+    property color chatViewModelListHighlightColor: {
```
MenuHighlightColor?

3Simplex commented:
Including the comboBox in the theme needed a little color change for the background of the settings menu, so it would look right. I used the scheme for the chatView to colorize the menu to adhere to the scheme within a chat.

![Screenshot 2024-07-08 191706](https://github.com/nomic-ai/gpt4all/assets/10260755/e63f6066-4a75-47ff-a505-d6654744ac24)
![Screenshot 2024-07-08 191716](https://github.com/nomic-ai/gpt4all/assets/10260755/c4bba37b-11cf-4481-ae56-a4d6ee58611c)
![Screenshot 2024-07-08 191734](https://github.com/nomic-ai/gpt4all/assets/10260755/48c30bb4-e175-4368-a6a2-2b2d4aeb4b2c)
![Screenshot 2024-07-08 191802](https://github.com/nomic-ai/gpt4all/assets/10260755/b7aa50bf-978e-47d7-b06a-630c2a4a324f)
![Screenshot 2024-07-08 191811](https://github.com/nomic-ai/gpt4all/assets/10260755/4ab6bddb-b977-4a61-9355-4c2ee15c4c4a)

manyoso commented:
This is fine by me, but asking our design guy Vincent to have a look to sign off

3Simplex commented:
Worked directly with Vincent, updated code for requested changes. 
- Colors for border modified, waiting on final color suggestion/approval for "Dark" mode. (The last approval required?)
- Frame removed, replaced with border. (per Vincent)
- Radius applied to selected menu items.  (per Vincent)
- Width of selected menu items adjusted to fit within menus, (No longer cuts off on right due to "clip: true")
- Removed the "rectangle" I added around the "Application Settings" (per Vincent)
- Found and connected the last remaining menu that was not included adhering to the new style. (ModelSettings Selection)

3Simplex commented:
All menus now follow these themes. 

"Dark" Awaiting final approval of Vincent. (Waiting...)
![Screenshot 2024-07-09 174825](https://github.com/nomic-ai/gpt4all/assets/10260755/0bcc52fb-0b9b-4088-8cf1-7ee204fd975d)

"Light" Awaiting final approval of Vincent (This looks darker/better in the app.)
![Screenshot 2024-07-09 174815](https://github.com/nomic-ai/gpt4all/assets/10260755/9f2f67a9-2f7d-41a0-9f26-c54f06f81cd7)

"LegacyDark" Approved by Vincent (Unless he changed his mind.)
![Screenshot 2024-07-09 174806](https://github.com/nomic-ai/gpt4all/assets/10260755/dc7ae668-127f-4be1-b991-dcbd463c39c5)

3Simplex commented:
Included fix for server text color in chatview. (tested)
![Screenshot 2024-07-09 203927](https://github.com/nomic-ai/gpt4all/assets/10260755/c5cd273e-150c-463b-aa27-6b54544aed8e)
![Screenshot 2024-07-09 204729](https://github.com/nomic-ai/gpt4all/assets/10260755/4b07cc7f-9dcc-4354-8834-161b45fbd798)
Other text is unaffected as expected.
![Screenshot 2024-07-09 204950](https://github.com/nomic-ai/gpt4all/assets/10260755/9cd0c28e-986a-4e52-9471-c3eeb06c4329)

giardiv commented:
Thank you @3Simplex !! Really cool work
